### DISCUSSION TITLE
Concurrency is missing in the language tour

### DISCUSSION THREAD

**Discussion by impredicative** [gleam-lang/gleam#3115] [Ideas & suggestions]
Given that Gleam makes claims of using the Erlang runtime, as a prospective user, I absolutely need to see all of the concurrency offerings of Erlang as made available in Gleam. To my surprise, **the language tour looks to cover nothing about concurrency**! Does Gleam even support Erlang-style concurrency? Why would someone even use an Erlang-like language if not for it? It would be desirable for the tour to cover all aspects of concurrency with clear examples. Without this, it makes zero sense to consider Gleam over Elixir.

For the time being, [GPT](https://chat.openai.com/share/4576a749-cf90-4e9e-b026-72a4de9a5fd3) is perhaps more useful in this regard. Looking around, I found https://hexdocs.pm/gleam_erlang/ but this serves a different purpose, and naturally is far less approachable than a tour guide can be.

**Comment by lpil** [comment]
Hello! The tour teaches the language, and concurrency is not part of the language. It is instead part of libraries.


Further it is not technically possible to include actor based concurrency in the tour as the tour runs on JavaScript and JavaScript cannot implement what Erlang can.

There will be more material on actors in future but the community decided that tooling is more important, so that is the focus of the core team currently.

**Please do not use GPT for this.** It is extremely inaccurate and will hallucinate aggressively for anything Gleam related. Instead ask in the Discord as there are many knowledgeable Gleam programmers ready to help you.

**Reply by impredicative** [reply]
Acknowledged. Speaking as a random developer, **concurrency is however the primary reason for using any Erlang ecosystem language**. It would be very valuable to have a comprehensive tutorial for it, even if it's separate from the tour. One shouldn't have to require Discord for something so essential.

**Reply by impredicative** [reply]
> the tour runs on JavaScript and JavaScript cannot implement what Erlang can

Well, that's quite the limitation. For such cases, it may help to run the code on a server instead and return the response. I think Go lang does it. It would be so beneficial for concurrency to be covered directly in the tour.

**Reply by lpil** [reply]
As I said, we will have comprehensive documentation on actors, but it is not appropriate to have it in the tour.

**Reply by inoas** [reply]
Maybe the final page could link to docs and options on concurrency once those are deemed good enough?
